addons/website_sale/views/snippets.xml

```diff
@@ -34,4 +34,71 @@
     </xpath>
 
 </template>
+
+<template id="remove_external_snippets" inherit_id="website.external_snippets">
+    <xpath expr="//t[@t-install='website_sale']" position="replace"/>
+</template>
+
+<template id="website_sale_snippet" inherit_id="website.snippets">
+    <xpath expr="//div[@id='snippet_feature']//t[@t-snippet][last()]" position="after">
+        <t t-snippet="website_sale.s_products_searchbar" t-thumbnail="/website/static/src/img/snippets_thumbs/s_products_searchbar.png"/>
+    </xpath>
+    <xpath expr="//div[@id='snippet_content']//t[@t-snippet][last()]" position="after">
+        <t t-snippet="website_sale.o_wsale_products_searchbar_input" t-thumbnail="/website/static/src/img/snippets_thumbs/s_products_searchbar.png"/>
+    </xpath>
+</template>
+
+<template id="website_sale_snippet_options" inherit_id="website.snippet_options">
+    <xpath expr="." position="inside">
+        <div data-js="ProductsSearchBar"
+            data-selector="form.o_wsale_products_search_bar_input">
+            <a tabindex="-1" href="#" class="dropdown-item" data-no-preview="true" data-open-searchbar-settings=""><i class="fa fa-search"></i>Products Search Bar</a>
+        </div>
+    </xpath>
+    <xpath expr="//div[@data-js='content']" position="attributes">
+        <attribute name="data-selector" add=".o_wsale_products_searchbar_input" separator=","/>
+        <attribute name="data-drop-near" add=".o_wsale_products_searchbar_input" separator=","/>
+    </xpath>
+</template>
+
+<template id="website_sale_search_box" inherit_id="website.website_search_box">
+    <xpath expr="//input[@name='search']" position="attributes">
+        <attribute name="class">search-query form-control o_wsale_search_products oe_search_box</attribute>
+        <attribute name="data-display-description">true</attribute>
+        <attribute name="data-display-price">true</attribute>
+        <attribute name="data-display-image">true</attribute>
+    </xpath>
+    <xpath expr="//div[hasclass('input-group')]" position="inside">
+        <input name="order" type="hidden" class="o_wsale_search_order_by" value="name asc"/>
+    </xpath>
+</template>
+
+<!-- Products Search Bar input-group template -->
+<template id="website_sale_search_input_group" name="Products Search Bar Input">
```

Comment:
In fact the two templates could be regrouped into one. This one should inherit from the website_search_box and wrap the content in the form. This can be done with `position="replace"` and `<t>$0</t>`to print the content being replaced. So:

**website.website_search_box**:
```xml
<template id="website.website_search_box">
    <div class="stuff">...</div>
</template>
```
**website_sale.website_sale_products_search_box**:
```xml
<template id="website.website_search_box">
    <xpath expr="..." position="attributes">...</xpath>
    <xpath expr="//div[hasclass('stuff')]" position="replace">
        <form ...>
             <t>$0</t>
             <t t-raw="0"/>
        </form>
    </xpath>
</template>
```